client/src/plugins/deployment-tool/DeploymentTool.js

```diff
@@ -41,17 +41,17 @@ export default class DeploymentTool extends PureComponent {
   }
 
   componentDidMount() {
-    this.props.subscribe('app.activeTabChanged', activeTab => {
+    this.props.subscribe('app.activeTabChanged', ({ activeTab }) => {
       this.setState({ activeTab });
     });
   }
 
-  saveTab() {
+  async saveTab(tab) {
```

Comment:
```suggestion
  saveTab(tab) {
```

We're returning a Promise anyway.